src/main/java/io/vlingo/xoom/schemata/query/SchemaQueriesActor.java

```diff
@@ -38,4 +41,34 @@ public SchemaQueriesActor(StateStore stateStore) {
         String reference = path.toReference();
         return queryStateFor(reference, NamedSchemaView.class);
     }
+
+    @Override
+    public Completes<NamedSchemaView> schemaByNamesWithRetries(String organization, String unit, String context,
+                                                               String schema, long retryInterval, int retryCount) {
+        return tryQuery(Tuple4.from(organization, unit, context, schema), completesEventually(), retryInterval, retryCount);
+    }
+
+    private Completes<NamedSchemaView> tryQuery(Tuple4<String, String, String, String> query,
+                                                CompletesEventually completes, long retryInterval, int remainingRetries) {
+        schemaByNames(query._1, query._2, query._3, query._4).andThenConsume(result -> {
+            if (result == null) {
+                retry(query, completes, retryInterval, remainingRetries);
+            } else {
+                completes.with(result);
+            }
+        });
+        return completes(); // note: this is NOT completesEventually
```

Comment:
For my understanding, why do we create a new `Completes` here instead of using the one that was passed in to the method?

Reply:
Looks like the `Completes` we return here will never be completed, so the client of `schemaByNamesWithRetries()` might wait for the outcome forever. I think we need to add a test verifying the outcome not only the count of retries.

Reply:
@jakzal It's not a new `Completes` or shouldn't be. The `CompletesEventually` wraps the `Completes` for pooling purposes. However, I might be misusing this now with the new `FutureCompletes` implementation. To your mind, is it safe to hold the `Completes` provided by `completes()` and also return it? If so, we might no longer need `CompletesEventually`, but we would have to pool differently.

The reason for retaining the same pooled `CompletesEventually` (actor) is so a given actor can't cause a later message to reach completion before a previous message completes. If the seems wrong, think completes pipeline (like a for comprehention).

Reply:
@VaughnVernon can you give me an example of such pooling?

Reply:
@jakzal It's in xoom-actors under pooled completes. See `io.vlingo.actors.plugins`. You should be able to find several examples throughout the platform. Search for `completesEventually()`.

Reply:
@jakzal Test case is failing. Do you have any idea why this is failing?

Reply:
I started by improving tests slightly.

Tests now verify if any outcome was actually returned. Previously no data was actually put into the state store.

Tests now also document the intent better. Instead of:

 ```
 queries.schemaByNames("OO", "UU", "CC", "SS")
 ```

 we now have:

 ```
 queries.schemaByNames("vlingo", "schemata-test", "io.vlingo.xoom.schemata", "SchemaDefined")
 ```

Abstract names were in a way of my understanding.

I will next take a look at how to make the test for `schemaByNamesWithRetries()` pass.